crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+use std::sync::RwLockWriteGuard;
+
+use crate::prelude::*;
+
+#[uniffi::export]
+impl SargonOS {
+    pub async fn analyse_transaction_preview(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        self.perform_transaction_preview_analysis(
+            instructions,
+            blobs,
+            message,
+            is_wallet_transaction,
+            nonce,
+            notary_public_key,
+        )
+        .await
+    }
+}
+
+impl SargonOS {
+    pub async fn perform_transaction_preview_analysis(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        let network_id = self.profile_state_holder.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        let transaction_manifest =
+            TransactionManifest::new(instructions, network_id, blobs)?;
+
+        // Get the transaction preview
+        let transaction_preview = self
+            .get_transaction_preview(
+                gateway_client,
+                transaction_manifest.clone(),
+                network_id,
+                message,
+                nonce,
+                notary_public_key,
+            )
+            .await?;
+        let engine_toolkit_receipt = transaction_preview
+            .radix_engine_toolkit_receipt
+            .ok_or(CommonError::FailedToExtractTransactionReceiptBytes)?;
```

Comment:
If we are gonna default to `FailedToExtractTransactionReceiptBytes`, then aren't we loosing the error mapped in `map_failed_transaction_preview()`?

Reply:
No, because the function will exit before, if `get_transaction_preview` has `Err` result.